spanner/tests/system.py

```diff
@@ -371,14 +445,19 @@ def test_transaction_read_and_insert_or_update_then_commit(self):
         rows = list(session.read(self.TABLE, self.COLUMNS, keyset))
         self._check_row_data(rows)
 
+    @staticmethod
+    def _row_data(max_index):
+        for index in range(max_index):
+            yield [
+                index,
+                'First%09d' % (index,),
+                'Last09%d' % (index),
+                'test-%09d@example.com' % (index,),
+            ]
+
     def _set_up_table(self, row_count):
         from google.cloud.spanner import KeySet
 
-        def _row_data(max_index):
-            for index in range(max_index):
-                yield [index, 'First%09d' % (index,), 'Last09%d' % (index),
-                       'test-%09d@example.com' % (index,)]
-
         keyset = KeySet(all_=True)
```

Comment:
See my allergy to dynamically defined functions [above][1]

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/3384#discussion_r115375627